.github/workflows/pull_request_secure.yml

```diff
@@ -259,6 +259,61 @@ jobs:
         env:
           PIP_FIND_LINKS: ${{ github.workspace }}/timefold-solver/dist
         run: tox
+
+  build_documentation:
+    runs-on: ubuntu-latest
+    needs: approval_required
+    name: Build Documentation
+    env:
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+    steps:
+      - name: Checkout frontend
+        id: checkout-frontend
+        uses: actions/checkout@v4
+        with:
+          repository: TimefoldAI/frontend
+          token: ${{ secrets.JRELEASER_GITHUB_TOKEN }} # Safe; only used to clone the repo and not stored in the fork.
+          fetch-depth: 0 # Otherwise merge will fail on account of not having history.
+
+      - name: Checkout timefold-solver
+        uses: actions/checkout@v4
+        with:
+          path: ./timefold-solver
+
+      - name: Set up NodeJs
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Build Documentation
+        env:
+          GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
+        run: |
+          cd ${{ github.workspace }}/timefold-solver/
+          GURL="$(git remote get-url origin)"
+          cd -
+          sed -i '/algolia/,+1d' apps/docs/antora-playbook.yml
+          sed -i '/timefold-orbit/,+3d' apps/docs/antora-playbook.yml
+          sed -i '/timefold-field/,+3d' apps/docs/antora-playbook.yml
+          sed -i '/timefold-models/,+3d' apps/docs/antora-playbook.yml
+          sed -i '/timefold-employee/,+3d' apps/docs/antora-playbook.yml
+          sed -i '/tags/,1d' apps/docs/antora-playbook.yml
+          sed -i "s#git@github.com:timefoldai/timefold-solver.git#$GURL#g" apps/docs/antora-playbook.yml
+          sed -i "s/branches:[ .-~]*/branches: $BRANCH_NAME/" apps/docs/antora-playbook.yml
```

Comment:
Instead of all these `sed` replaces, I'd prefer something like the below for a couple of reasons:
- It's independent of any changes made to the original file (right now, making changes in the sources of the original file, might break your sed-replace
- It doesn't give any insights into our private repo / setup
The following code just clears out the whole `sources` section and only puts the relevant solver source in there:

```awk -vGURL="${GURL}" -vBRANCH_NAME="${BRANCH_NAME}" 'BEGIN{ print_flag=1 }
{
    if( $0 ~ /  sources:/ )
    {
       print "  sources:"
       print "    - url: ", GURL
       print "      branches: [",BRANCH_NAME,"]"
       print "      start_path: docs/src"
       print_flag=0;
       next
    }
    if( $0 ~ /^$/ )
    {
        print_flag=1;
    }
    if ( print_flag == 1 )
        print $0

} ' antora-playbook.yml

Reply:
I found an easier way to do this